### PR TITLE
Simplify Dockerfile

### PR DESCRIPTION
**Pull request by corny**

### CODE REVIEW

File: musicbrainz-dockerfile/Dockerfile [hunk 1]
```diff
@@ -28,9 +27,8 @@ RUN curl -sSL https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
         python-minimal \
         zlib1g-dev
 
-RUN git clone https://github.com/metabrainz/musicbrainz-server.git musicbrainz-server && \
-    cd musicbrainz-server && \
-    git checkout v-2020-04-13
+ARG MUSICBRAINZ_VERSION=v-2020-04-13
+RUN git clone https://github.com/metabrainz/musicbrainz-server.git -c advice.detachedHead=false -b $MUSICBRAINZ_VERSION musicbrainz-server
```
Comment (SuperSandro2000) [inline musicbrainz-dockerfile/Dockerfile]:
```suggestion
RUN git clone https://github.com/metabrainz/musicbrainz-server.git -c advice.detachedHead=false -b $MUSICBRAINZ_VERSION --depth=1 musicbrainz-server
```

I don't think we need the entire history.

**Review by SuperSandro2000 — COMMENTED** [review]
.

File: musicbrainz-dockerfile/Dockerfile (hunk 1 above)
Comment (corny) [reply musicbrainz-dockerfile/Dockerfile]: Good point. I didn't know this option before. I just added it to all clone commands.

File: musicbrainz-dockerfile/Dockerfile [hunk 2]
```diff
@@ -2,9 +2,8 @@ FROM metabrainz/base-image
 LABEL maintainer=jeffsturgis@gmail.com
 
 ENV DOCKERIZE_VERSION v0.5.0
-RUN wget https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && \
-    tar -C /usr/local/bin -xzvf dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && \
-    rm -f dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz
+RUN curl -sSL https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz | \
+    tar -C /usr/local/bin -xzv
```
Comment (yvanzo) [inline musicbrainz-dockerfile/Dockerfile:6]: This requires to set the shell option `pipefail` to reliably detect failures, as in https://github.com/metabrainz/musicbrainz-docker/blob/mbvm-38-dev/build/musicbrainz/Dockerfile#L10

File: musicbrainz-dockerfile/Dockerfile [hunk 3]
```diff
@@ -28,9 +27,8 @@ RUN curl -sSL https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
         python-minimal \
         zlib1g-dev
 
-RUN git clone https://github.com/metabrainz/musicbrainz-server.git musicbrainz-server && \
-    cd musicbrainz-server && \
-    git checkout v-2020-04-13
+ARG MUSICBRAINZ_VERSION=v-2020-04-13
+RUN git clone --depth=1 https://github.com/metabrainz/musicbrainz-server.git -c advice.detachedHead=false -b $MUSICBRAINZ_VERSION musicbrainz-server
```
Comment (yvanzo) [inline musicbrainz-dockerfile/Dockerfile:31]: Same change as https://github.com/metabrainz/musicbrainz-docker/commit/28fd525454582d75287a08abae7c6852e0eeeef7 available in the development branch [`mbvm-38-dev`](https://github.com/metabrainz/musicbrainz-docker/tree/mbvm-38-dev) already.

File: musicbrainz-dockerfile/Dockerfile [hunk 4]
```diff
@@ -64,13 +62,7 @@ RUN eval $( perl -Mlocal::lib) && cpanm --notest \
...
 
 ADD DBDefs.pm /musicbrainz-server/lib/
-ADD scripts/start.sh /start.sh
-ADD scripts/start_mb_renderer.pl /start_mb_renderer.pl
-ADD scripts/crons.conf /crons.conf
-ADD scripts/replication.sh /replication.sh
-ADD scripts/createdb.sh /createdb.sh
-ADD scripts/recreatedb.sh /recreatedb.sh
-ADD scripts/set-token.sh /set-token.sh
+ADD scripts/* /
```
Comment (yvanzo) [inline musicbrainz-dockerfile/Dockerfile:65]: Same change as https://github.com/metabrainz/musicbrainz-docker/commit/03f749abf47aff36234e29d60ec1a2f43ffc1a90 available in the development branch [`mbvm-38-dev`](https://github.com/metabrainz/musicbrainz-docker/tree/mbvm-38-dev) already.

File: postgres-dockerfile/Dockerfile [hunk 5]
```diff
@@ -20,8 +20,8 @@ RUN apt-get update && \
         pkg-config \
         postgresql-server-dev-${POSTGRES_VERSION}
 
-RUN git clone https://github.com/metabrainz/postgresql-musicbrainz-unaccent.git /tmp/postgresql-musicbrainz-unaccent && \
-    git clone https://github.com/metabrainz/postgresql-musicbrainz-collate.git /tmp/postgresql-musicbrainz-collate
+RUN git clone --depth=1 https://github.com/metabrainz/postgresql-musicbrainz-unaccent.git /tmp/postgresql-musicbrainz-unaccent && \
+    git clone --depth=1 https://github.com/metabrainz/postgresql-musicbrainz-collate.git /tmp/postgresql-musicbrainz-collate
```
Comment (yvanzo) [inline postgres-dockerfile/Dockerfile:24]: These changes should apply to https://github.com/metabrainz/musicbrainz-docker/blob/mbvm-38-dev/build/postgres/Dockerfile#L25-L27 instead.

**Review by yvanzo — CHANGES_REQUESTED** [review]
Thanks. The `master` branch is currently in maintenance mode only, none of these changes is actually needed to keep working existing servers, thus won’t be merged. Half of these changes have already been made to the development branch [`mbvm-38-dev`](https://github.com/metabrainz/musicbrainz-docker/tree/mbvm-38-dev). For the other half, please rebase your pull request on that branch instead. (Or I can do it if you prefer.)

File: musicbrainz-dockerfile/Dockerfile (hunk 3 above)
Comment (corny) [reply musicbrainz-dockerfile/Dockerfile:31]: Good to know, thank you!

File: musicbrainz-dockerfile/Dockerfile (hunk 2 above)
Comment (corny) [reply musicbrainz-dockerfile/Dockerfile:6]: Good to know, thank you!

**Comment by corny** [comment]
I close this PR and I will check out the development branch.

**Pull request closed without merging**